components/CharmCalculator.tsx

```diff
@@ -0,0 +1,198 @@
+   *
+   * @param chosenClass | String
+   */
+  function setClass(chosenClass: String): void {
+    // set class, then set available heroes based on that class
+    setSelectedClass(chosenClass);
+    setSelectedHero(undefined);
+
+    Heroes.Classes.find((classType) => {
+      if (classType.Class === chosenClass) {
+        setAvailableHeroes(classType.Heroes.map((hero) => hero.Name));
+      }
+    });
+  }
+
+  /** Sets the current hero and clears the selected charms
+   *  Sets the available charms based on selected hero
+   *
+   * @param heroName | String
+   */
+  function setHero(heroName: String): void {
+    // set hero, then get list of available charms based on that hero
+    setSelectedHero(heroName);
+    setSelectedLeftCharm(undefined);
+    setSelectedMidCharm(undefined);
+    setSelectedRightCharm(undefined);
+
+    Heroes.Classes.find((classType) => {
+      if (classType.Class === selectedClass) {
+        classType.Heroes.find((hero) => {
+          if (hero.Name === heroName) {
+            let charms: Charm[] = [];
+            hero.Charms.forEach((charm) => {
+              charms.push(new Charm(charm.Name, charm.Stats));
+            });
+            setAvailableCharms(charms);
+          }
+        });
+      }
+    });
+  }
+
+  /** Sets the charm that you selected for the given position
+   *  Any time a charm is changed, recalculate the summary
+   *
+   * @param charmName | String
+   * @param position | CharmPosition (enum) | Left/Mid/Right
+   */
+  function setCharm(charmName: string, position: CharmPosition): void {
+    // Get charm and set that selectedCharm to the current charm.
+    availableCharms?.find((charm) => {
+      if (charm.name === charmName) {
+        switch (position) {
+          case CharmPosition.Left:
+            setSelectedLeftCharm(charm);
+            setStatsSummary(GetStatSummary(charm, selectedMidCharm, selectedRightCharm));
+            break;
+          case CharmPosition.Mid:
+            setSelectedMidCharm(charm);
+            setStatsSummary(GetStatSummary(selectedLeftCharm, charm, selectedRightCharm));
+            break;
+          case CharmPosition.Right:
+            setSelectedRightCharm(charm);
+            setStatsSummary(GetStatSummary(selectedLeftCharm, selectedMidCharm, charm));
+            break;
+          default:
+            break;
+        }
+      }
+    });
+  }
+
+  return (
+    <>
+      <div id="options" className="flex flex-row pt-8 gap-x-2">
+        <Select onValueChange={(e) => setClass(e)}>
+          <SelectTrigger>
+            <SelectValue placeholder="Class" />
+          </SelectTrigger>
+          <SelectContent>
+            {availableClasses.map((type) => (
+              <SelectItem value={type} key={type}>
+                {type}
+              </SelectItem>
+            ))}
+          </SelectContent>
+        </Select>
+        <Select disabled={!availableHeroes} onValueChange={(e) => setHero(e)}>
```

Comment:
to make this go back to the placeholder when the value is unset:

```jsx
<Select value={selectedHero}>
```

When the attribute is `null` or `undefined` it should fallback appropriately.

[This](https://www.radix-ui.com/primitives/docs/components/select#api-reference) is the underlying component that `shadcn/ui` styles and wraps.